app/src/main/java/com/asfoundation/wallet/topup/payment/AdyenTopUpPresenter.kt

```diff
@@ -49,20 +49,36 @@ class AdyenTopUpPresenter(private val view: AdyenTopUpView,
 ) {
 
   private var waitingResult = false
+  private var paymentMethod = PaymentType.CARD.name
 
   fun present(savedInstanceState: Bundle?) {
     if (savedInstanceState != null) {
       waitingResult = savedInstanceState.getBoolean(WAITING_RESULT)
     }
     loadPaymentMethodInfo(savedInstanceState)
-    handleErrorDismissEvent()
     handleForgetCardClick()
 
+    handleRetryClick(savedInstanceState)
     handleRedirectResponse()
     handleSupportClicks()
     handleTryAgainClicks()
   }
 
+  private fun handleRetryClick(savedInstanceState: Bundle?) {
+    disposables.add(view.retryClick()
+        .observeOn(viewScheduler)
+        .doOnNext { view.showRetryAnimation() }
+        .delay(1, TimeUnit.SECONDS)
+        .doOnNext {
+          if (waitingResult) {
+            view.navigateToPaymentSelection()
```

Comment:
We should navigate to the transactionActivity in this case. (Idk if this was what we were talking before). In the transactionActivity at least you can check if you already received the topup before trying again